packages/auth/amplify_auth_cognito_dart/lib/src/auth_plugin_impl.dart

```diff
@@ -269,6 +268,31 @@ class AmplifyAuthCognitoDart extends AuthPluginInterface<
       );
   }
 
+  @protected
+  Future<void> registerAnalyticsMetadata({
+    required AmplifyConfig config,
+    LegacyNativeDataProvider? dataProvider,
+  }) async {
+    // Setup AnalyticsMetadataType
+    final analyticsConfig = config.auth?.awsPlugin?.pinpointAnalytics?.default$;
+    if (analyticsConfig != null) {
+      final appId = analyticsConfig.appId;
+
+      final storageFactory = _stateMachine.expect<SecureStorageFactory>();
+      final endpointStore = storageFactory(scope: 'analytics');
+      final endpointStoreManager = EndpointInfoStoreManager(
+          store: endpointStore, legacyNativeDataProvider: dataProvider);
+
+      await endpointStoreManager.init(pinpointAppId: appId);
```

Comment:
I think this whole method should ideally be done in the `ConfigurationStateMachine`. The only thing we want to do in `plugin_impl` is add everything to the dependency manager so that this method only gets called once.